TicketSwap/solar
# With useTransition hook

# Description

This PR adds a custom `useTransition()` hook to the library and replaces `react-spring` in `<Dialog />` and `<MenuButton />`.

## How to test

- Checkout this branch
- `$ yarn dev`
- Visit http://localhost:9001/?path=/story/dialog--basic
- Verify the mount/unmount transition works as expected
- Visit http://localhost:9001/?path=/story/menubutton--default
- Verify the mount/unmount transition works as expected
